proxy/describe.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) OpenFaaS Author(s) 2018. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package proxy
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/openfaas/faas/gateway/requests"
+)
+
+//GetFunctionInfo get an OpenFaaS function information
+func GetFunctionInfo(gateway string, functionName string, tlsInsecure bool) (requests.Function, error) {
+	var result requests.Function
+
+	gateway = strings.TrimRight(gateway, "/")
```

Comment:
Should we use `url.Parse` here and then use `path.Join` to assemble the endpoint to ensure we get a clean url?